test/e2e/usersignup_test.go

```diff
@@ -45,662 +43,190 @@ func (s *userSignupIntegrationTest) TearDownTest() {
-
-	// Create user signup - no approval set
-	s.T().Logf("Creating UserSignup with namespace %s", s.namespace)
-
-	userSignup, err := newUserSignup(s.awaitility.Host(), uuid.NewV4().String(), "reginald@alpha.com")
-	require.NoError(s.T(), err)
+	userSignup := newUserSignup(s.T(), s.awaitility.Host(), uuid.NewV4().String(), "reginald@alpha.com")
 
 	// Remove the specified target cluster
 	userSignup.Spec.TargetCluster = ""
-	err = s.awaitility.Client.Create(context.TODO(), userSignup, testsupport.CleanupOptions(s.testCtx))
+	err := s.awaitility.Client.Create(context.TODO(), userSignup, testsupport.CleanupOptions(s.testCtx))
 	require.NoError(s.T(), err)
 	s.T().Logf("user signup '%s' created", userSignup.Name)
 
-	// Confirm the UserSignup was created
-	_, err = s.hostAwait.WaitForUserSignup(userSignup.Name)
-	require.NoError(s.T(), err)
-
-	// Get the newly created UserSignup resource
-	userSignup, err = s.hostAwait.WaitForUserSignup(userSignup.Name)
-	require.NoError(s.T(), err)
-
-	// Confirm the MasterUserRecord was created
-	_, err = s.hostAwait.WaitForMasterUserRecord(userSignup.Status.CompliantUsername)
-	require.NoError(s.T(), err)
-
-	// Confirm that:
-	// 1) the Approved condition is set to true
-	// 2) the Approved reason is set to ApprovedAutomatically
-	// 3) the Complete condition is (eventually) set to true
-	_, err = s.hostAwait.WaitForUserSignup(userSignup.Name, wait.UntilUserSignupHasConditions(
-		v1alpha1.Condition{
-			Type:   v1alpha1.UserSignupApproved,
-			Status: corev1.ConditionTrue,
-			Reason: "ApprovedAutomatically",
-		},
-		v1alpha1.Condition{
-			Type:   v1alpha1.UserSignupComplete,
-			Status: corev1.ConditionTrue,
-		}))
+	// Check the UserSignup is approved now
+	userSignup, err = s.hostAwait.WaitForUserSignup(userSignup.Name, wait.UntilUserSignupHasConditions(approvedAutomatically()...))
 	require.NoError(s.T(), err)
 
-	// Lookup the MUR
-	mur := &v1alpha1.MasterUserRecord{}
-	err = s.awaitility.Client.Get(context.TODO(), types.NamespacedName{Namespace: s.namespace, Name: userSignup.Status.CompliantUsername}, mur)
-	require.NoError(s.T(), err)
-
-	require.Len(s.T(), mur.Spec.UserAccounts, 1)
-
-	// Confirm the target cluster was set
-	require.NotEmpty(s.T(), mur.Spec.UserAccounts[0].TargetCluster)
+	// Confirm the MUR was created and target cluster was set
+	s.assertCreatedMUR(userSignup)
 }
 
-func (s *userSignupIntegrationTest) TestUserSignupWithAutoApprovalNoApprovalSet() {
-	// Set the user approval policy to automatic
-	s.setApprovalPolicyConfig("automatic")
-
-	// Create user signup - no approval set
-	s.T().Logf("Creating UserSignup with namespace %s", s.namespace)
+func (s *userSignupIntegrationTest) TestTransformUsername() {
+	// Create UserSignup with a username that we don't need to transform
+	userSignup, _ := s.createUserSignupWithNameAndAssertManualApproval(true, "paul-no-need-to-transform")
+	require.Equal(s.T(), "paul-no-need-to-transform", userSignup.Status.CompliantUsername)
```

Comment:
To be honest, this test looks like a unit test. The only thing that is verified here is that the compliant username is set in the status of the usersignup - the rest is verified in other test cases. I believe that if we use verify the generation of the  compliant username in the previous test `TestTargetClusterSelectedAutomatically`, then we should be fine and we can drop this entire test

Reply:
It does test more. Not just the fact that compliant username is set. It tests all possible combinations:
- username == compliant username
- there is already a username with such compliant username
- there are multiple matches (so we should keep increasing the number)

We could merge it with other tests but it would make the test code a bit less readable IMO.